community/aqua/chemistry/QSE_pytket.ipynb

```diff
@@ -0,0 +1,364 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Set up initial state and variational form <a id='sectionB'></a>\n",
+    "\n",
+    "We define the molecular hamiltonian and map it to a qubit operator according to some appropriate scheme such as the Jordan-Wigner transform. We initialize our initial state (a Hartree-Fock state) and our variational ansatz (a unitary coupled cluster with single and double excitations, UCCSD)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qiskit_aqua_chemistry import FermionicOperator\n",
+    "from qiskit_aqua_chemistry.aqua_extensions.components.initial_states import HartreeFock\n",
+    "from qiskit_aqua_chemistry.aqua_extensions.components.variational_forms import UCCSD\n",
+    "\n",
+    "n_qubits = molecule._one_body_integrals.shape[0]\n",
+    "n_electrons = molecule._num_alpha + molecule._num_beta - molecule._molecular_charge\n",
+    "\n",
+    "# get fermionic operator and mapping to qubit operator\n",
+    "ferOp = FermionicOperator(h1=molecule._one_body_integrals, h2=molecule._two_body_integrals)\n",
+    "\n",
+    "qubitOp = ferOp.mapping(map_type='JORDAN_WIGNER', threshold=0.00000001)\n",
+    "n_qubits = qubitOp.num_qubits\n",
+    "qubitOp.chop(10**-10)\n",
+    "\n",
+    "# setup variation form generator (generate trial circuits for VQE)\n",
+    "initial_hf = HartreeFock(num_qubits=n_qubits, num_orbitals=n_qubits, \n",
+    "                    qubit_mapping='jordan_wigner', two_qubit_reduction=False, num_particles= n_electrons)\n",
+    "\n",
+    "var_form = UCCSD(num_qubits=n_qubits, num_orbitals=n_qubits, \n",
+    "                num_particles=n_electrons, depth=1, initial_state=initial_hf, qubit_mapping='jordan_wigner')\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Use VQE to find ground state <a id='sectionB'></a>\n",
+    "\n",
+    "We run the VQE algorithm with the components previously defined. The calculation provides us with the ground state energy and optimized parameters for the corresponding UCCSD wavefunction. In this example we use the L_BFGS_B algorithm for classical optimization."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "GS Minimum value: -1.1361894540653719\n",
+      "GS Parameters: [ 5.25314574e-07  5.11815141e-07 -1.04867342e-01]\n"
+     ]
+    }
+   ],
+   "source": [
+    "from qiskit_aqua.algorithms.components.optimizers import L_BFGS_B\n",
+    "from qiskit_aqua.algorithms.adaptive import VQE\n",
+    "\n",
+    "\n",
+    "# set initial values of parameters\n",
+    "number_amplitudes = len(var_form._single_excitations)+ len(var_form._double_excitations)\n",
```

Comment:
if you just want to get the number of parameters in the variational form, you can do `var_form.num_parameters` but I guess you would like to convey the number of amplitude in single and double excitations.

Reply:
yes would rather keep that in